### PR TITLE
Redo app boilerplate and add compatibility with Django up to 4.1

### PR DESCRIPTION
**Pull request by ewjoachim**
Hi there,

It appears that this project is used in some project I use at work, and I'm trying to upgrade my project to recent Django versions. It looks like this project is incompatible with Django >= 3.1 due to a change made in Django by a certain individual named @David-Wobrock (https://github.com/django/django/pull/12518/commits/71c1b7fb34109e6270841eb5d2dcd60f8a22453b).

This PR:
- Fixes the issue while staying compatible with Django before and after the change
- Clarifies the compatible versions in the classifiers
- Add lint & formatting through pre-commit and the usual suspects
- Removes Travis CI configuration and add GitHub Actions, setup GitHub Workflow for auto-releasing the library on PyPI when a tag is released (e.g. through the Releases tab in GitHub)(note: for this to work, setup a PYPI_TOKEN secret on the repo)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) <year> <copyright holders>
+Copyright (c) 2017- Jean-Baptiste Barth - Botify
```
Comment (David-Wobrock) [inline LICENSE:3]: Look at that! @jbbarth
Good ol' times 🧓

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  hooks:
+  - id: flake8
+    exclude: "tests/snapshots/"
+
+- repo: https://github.com/asottile/pyupgrade
+  rev: "v3.3.1"
+  hooks:
+  - id: pyupgrade
+    args: [ --py37-plus ]
+    exclude: "tests/snapshots/"
```
Comment (David-Wobrock) [inline .pre-commit-config.yaml:35]: No [ruff](https://github.com/charliermarsh/ruff) yet? 😏

**Review by David-Wobrock — COMMENTED** [review]
> due to a change made in Django by a certain individual named @David-Wobrock (https://github.com/django/django/commit/71c1b7fb34109e6270841eb5d2dcd60f8a22453b).

Jeez', that guy 🙄 He's the worst.
Always breaking APIs and _usually_ getting away with it 😜 🙈

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ewjoachim) [reply .pre-commit-config.yaml:35]: TIL 😮  seems neat !

**Comment by ewjoachim** [comment]
btw @jbbarth I think you're the only one with maintainer rights on this lib on PyPI.

- Either you can add me (`ewjoachim`) in the project maintainers PyPI (with the permission to add other people)
- Or take a vow to deal with this project properly in the future should we need it (and securely transmit us a pypi token scoped to this project) (you'll find my gpg key at https://github.com/ewjoachim.gpg)
- Or if you want to take it and maintain it on your side, maybe we can discuss moving the GitHub repo to your space ?

(hm... Forgot to mention I work at Botify. But given I'm not a member of `@botify-labs` yet, you have no reason to just believe me. Let me take care of that first :D )

**Comment by ewjoachim** [comment]
> Jeez', that guy roll_eyes He's the worst.

BTW, I just saw you made quite a few impressive contributions to Django, congratulations :)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (David-Wobrock) [reply .pre-commit-config.yaml:35]: It's a new trend in Python community!
Here's an interesting take from the ruff creator, if you have a few minutes: https://notes.crmarsh.com/ruff-the-first-200-releases

File: tests/test_app.py [hunk 3]
```diff
@@ -16,7 +19,7 @@
     "tests.example_app",
 ]
 
-ROOT_URLCONF = "tests.urls"
+ROOT_URLCONF = "tests.settings"
```
Comment (ewjoachim) [inline tests/test_app.py:22]: test_app

File: tests/test_app.py [hunk 4]
```diff
@@ -31,3 +34,11 @@
         "NAME": os.path.join(BASE_DIR, "db.sqlite3"),
     }
 }
+
+urlpatterns = []
+
+if __name__ == "__main__":
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "tests.settings")
```
Comment (ewjoachim) [inline tests/test_app.py:43]: tests.test_app

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -4,16 +4,70 @@ on:
...
+
+          - name: Python 3.10 / Django 4.0
+            python_version: "3.10"
+            django_version: "4.0.*"
+
+          - name: Python 3.11 / Django 4.1
+            python_version: "3.11"
+            django_version: "4.1.*"
+
+    name: "py${{ matrix.python_version }}"
```
Comment (ewjoachim) [inline .github/workflows/ci.yml:43]: display django version

File: .github/workflows/release-drafter.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: Release Drafter
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
```
Comment (ewjoachim) [inline .github/workflows/release-drafter.yml:8]: permission

**Review by SuperDormito — APPROVED** [review]

**Pull request merged**